### ISSUE TITLE
[ORACLE] Reference Partition migration is not supported

### ISSUE THREAD

**Issue by chetank-yb** [yugabyte/yb-voyager#454]
Description : "WARNING: unsupported partition type on table 'ORDER_ITEMS'" thrown on attempting to export a schema which contains a reference partition

**SCHEMA**
```
CREATE TABLE orders
    ( order_id           NUMBER(12),
      order_date         DATE,
      order_mode         VARCHAR2(8),
      customer_id        NUMBER(6),
      order_status       NUMBER(2),
      order_total        NUMBER(8,2),
      sales_rep_id       NUMBER(6),
      promotion_id       NUMBER(6),
      CONSTRAINT orders_pk PRIMARY KEY(order_id)
    )
  PARTITION BY RANGE(order_date)
    ( PARTITION Q1_2005 VALUES LESS THAN (TO_DATE('01-APR-2005','DD-MON-YYYY')),
      PARTITION Q2_2005 VALUES LESS THAN (TO_DATE('01-JUL-2005','DD-MON-YYYY')),
      PARTITION Q3_2005 VALUES LESS THAN (TO_DATE('01-OCT-2005','DD-MON-YYYY')),
      PARTITION Q4_2005 VALUES LESS THAN (TO_DATE('01-JAN-2006','DD-MON-YYYY'))
    );

CREATE TABLE order_items
    ( order_id           NUMBER(12) NOT NULL,
      line_item_id       NUMBER(3)  NOT NULL,
      product_id         NUMBER(6)  NOT NULL,
      unit_price         NUMBER(8,2),
      quantity           NUMBER(8),
      CONSTRAINT order_items_fk
      FOREIGN KEY(order_id) REFERENCES orders(order_id)
    )
    PARTITION BY REFERENCE(order_items_fk);
```

**Steps to recreate**
1. Export the env variables for source and target.
2.  Run the export schema command as follows
     ```yb-voyager export schema --export-dir ${EXPORT_DIR} \
        --source-db-type ${SOURCE_DB_TYPE} \
        --source-db-host ${SOURCE_DB_HOST} \
        --source-db-user ${SOURCE_DB_USER} \
        --source-db-password ${SOURCE_DB_PASSWORD} \
        --source-db-name ${SOURCE_DB_NAME} \
        --source-db-schema ${SOURCE_DB_SCHEMA} --start-clean --yes```



![Image](https://user-images.githubusercontent.com/100332069/195330788-916555a3-a4c0-447e-b851-e63429f3023d.png)

**Comment by rahulb-yb** [comment]
Translated schema:
```
CREATE TABLE orders (
	order_id bigint NOT NULL,
	order_date timestamp,
	order_mode varchar(8),
	customer_id integer,
	order_status smallint,
	order_total double precision,
	sales_rep_id integer,
	promotion_id integer,
	PRIMARY KEY (order_id)
) PARTITION BY RANGE (order_date) ;


CREATE TABLE order_items (
	order_id bigint NOT NULL,
	line_item_id smallint NOT NULL,
	product_id integer NOT NULL,
	unit_price double precision,
	quantity integer
) -- Unsupported partition type, please check
 ;
ALTER TABLE order_items ALTER COLUMN ORDER_ID SET NOT NULL;
ALTER TABLE order_items ALTER COLUMN LINE_ITEM_ID SET NOT NULL;
ALTER TABLE order_items ALTER COLUMN PRODUCT_ID SET NOT NULL;

```
Note that despite the warning, the schema export goes through. We can easily catch this error in the DDL and report it during analyse-schema phase; no need to separately document this either, partition based on referenced keys is not a feature native to PG/YB.

TLDR: No longer a documentation item, report this partition type as unsupported in analyse-schema

**Comment by rahulb-yb** [comment]
Closing this issue as we already have a check for the keyword "unsupported" in a comment,  in our analyze-schema functionality
